File: common/src/main/kotlin/net/impleri/playerskills/events/handlers/CommandEvents.kt
package net.impleri.playerskills.events.handlers

import com.mojang.brigadier.CommandDispatcher
import dev.architectury.event.events.common.CommandRegistrationEvent
import net.impleri.playerskills.commands.PlayerSkillsCommands
import net.minecraft.commands.CommandBuildContext
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands

internal class CommandEvents {
  fun registerEvents() {
    CommandRegistrationEvent.EVENT.register(
      CommandRegistrationEvent { dispatcher: CommandDispatcher<CommandSourceStack>, _: CommandBuildContext, _: Commands.CommandSelection ->
        PlayerSkillsCommands.register(dispatcher)
      },
    )
  }
}

File: common/src/main/scala/net/impleri/playerskills/server/bindings/CommandEvents.scala
package net.impleri.playerskills.server.bindings

import dev.architectury.event.events.common.CommandRegistrationEvent
import dev.architectury.event.Event
import net.impleri.playerskills.server.commands.PlayerSkillsCommands

case class CommandEvents(event: Event[CommandRegistrationEvent] = CommandRegistrationEvent.EVENT) {
  def registerEvents(commands: PlayerSkillsCommands): Unit = {
    event.register((d, _, _) => commands.register(d))
  }
}
